### PR TITLE
Multiarch build server docker images

### PR DESCRIPTION
**Pull request by 6543**

### CODE REVIEW

File: .woodpecker/docker.yml [hunk 1]
```diff
@@ -1,87 +1,92 @@
...
-    secrets: [ docker_username, docker_password ]
     settings:
       dry_run: true
       repo: woodpeckerci/woodpecker-server
-      dockerfile: docker/Dockerfile.server.alpine
+      dockerfile: docker/Dockerfile.server.alpine.multiarch
+      platforms: linux/arm/v7,linux/arm64/v8,linux/amd64,linux/ppc64le
       tag: next-alpine
     when:
+      branch: ${CI_REPO_DEFAULT_BRANCH}
```
Comment (6543) [inline .woodpecker/docker.yml:41]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -88,6 +95,25 @@ build: build-agent build-server build-cli
 
 release-frontend: build-frontend
 
+check-xgo:
+	@hash xgo > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
+		$(GO) install src.techknowlogick.com/xgo@latest; \
+	fi
+
+cross-compile-server: cross-compile-server-build-loop
+	tree dist
```
Comment (relnod) [inline Makefile:104]: Hi, could you just run `tree` to see if the binaries are placed outside of the `dist/` folder?

File: Makefile [hunk 3]
```diff
@@ -88,6 +95,25 @@ build: build-agent build-server build-cli
...
+	@$(foreach platform,$(subst ;, ,$(PLATFORMS)),TARGETOS=$(firstword $(subst |, ,$(platform))) TARGETARCH=$(word 2,$(subst |, ,$(platform))) make normalize-server-artifacts || exit 1;)
+
+cross-compile-server-build-loop:
+	$(foreach platform,$(subst ;, ,$(PLATFORMS)),TARGETOS=$(firstword $(subst |, ,$(platform))) TARGETARCH=$(word 2,$(subst |, ,$(platform))) make release-server-xgo || exit 1;)
+
+normalize-server-artifacts:
+	mv dist/server/$(TARGETOS)/$(TARGETARCH)/$(shell ls dist/server/$(TARGETOS)/$(TARGETARCH)/) dist/server/$(TARGETOS)/$(TARGETARCH)/woodpecker-server
```
Comment (relnod) [inline Makefile:111]: For me it only works when I run the following here:
```
mv dist/server/$(TARGETOS)/$(TARGETARCH)/woodpecker-server-$(TARGETOS)-$(TARGETARCH) dist/server/$(TARGETOS)/$(TARGETARCH)/woodpecker-server

```

**Comment by relnod** [comment]
Hi @6543,
thanks for working on this!

I also can't reproduce this locally using `make cross-compile-server`. I can however reproduce it using the following Dockerfile:
```Dockerfile
FROM techknowlogick/xgo:go-1.17.x

WORKDIR /src
COPY . .

RUN apt update
RUN apt install -y tree
RUN PLATFORMS="linux|amd64;linux|arm64" TAGS="bindata sqlite sqlite_unlock_notify" make cross-compile-server

```

File: Makefile (hunk 2 above)
Comment (relnod) [reply Makefile:104]: Binaries are placed in `/build/`

**Comment by 6543** [comment]
->  #821

**Pull request closed without merging**